File: TwoPointers/SquaresofaSortedArray.java
package TwoPointers;

public class SquaresofaSortedArray {

    public static int[]sortedSquares(int[] nums){

        int left = 0;
        int right = nums.length -1 ;
        int[] result = new int[nums.length];
        int index= nums.length -1;

        while (left < right){
            if (Math.abs(nums[left]) < Math.abs(nums[right])){
                result[index--] = nums[left] * nums[left];
                left++;
            } else {
                result[index--] = nums[right] * nums[right];
                right--;
            }
        }
        return result;

    }
    public static void main(String[] args){

        int[] nums = {-4,-1,0,3,10};
        System.out.println(sortedSquares(nums));
    } 
}

File: TwoPointers/SquaresofaSortedArray.py
class Solution:
    def sortedSquares(self, nums: list[int]) -> list[int]:
    #    squaredList = [n ** 2 for n in nums]
        left = 0
        right = len(nums) -1
        result = []
       
        while left <= right:
           if nums[left] ** 2 > nums[right] ** 2:
               result.append(nums[left] ** 2)
               left += 1
           else:
               result.append(nums[right] ** 2)
               right += 1
          
               
        result.reverse()
        return result
        












if __name__ == "__main__":
    nums = [-4,-1,0,3,10]
    obj = Solution()
    print(obj.sortedSquares(nums))